src/config.rs

```diff
@@ -90,13 +89,29 @@ pub enum Email {
 /// ## Notes
 ///
 /// This function handles testing and non-testing environments.
-pub fn config_path() -> Result<PathBuf> {
+#[derive(Error, Debug)]
+pub enum ConfigError {
+    #[error("Failed to get home directory")]
+    HomeDir,
+    #[error("Failed to create config directory")]
+    CreateConfigDir,
+    #[error("Failed to create config file")]
+    CreateConfigFile,
```

Comment:
```suggestion
    #[error(transparent)]
    CreateFile(#[from] std::io::Error), 
```

Check the `thiserror` docs (https://docs.rs/thiserror/latest/thiserror/) on how to use these conversion macros.

Additionally, this should be `CreateFile` since it is already inside the `ConfigError` it is quite obvious that it is the "Config" file that we are talking about.